extensions/skins/conversation_v1/Conversation.js

```diff
@@ -75,227 +218,425 @@ oppia.directive('conversationSkin', [function() {
+        }
+        if (!$scope.interactionIsInline) {
+          $scope.panels.push($scope.PANEL_INTERACTION);
+        }
+        $scope.resetVisiblePanel();
+      };
 
-          // Do not scroll up.
-          // This occurs if Oppia gives no feedback for (e.g.) a supplemental
-          // interaction. This leads to a scroll *up* to Oppia's last output,
-          // which is rather disconcerting.
-          if ($(document).scrollTop() >= newScrollTop) {
-            newScrollTop = $(document).scrollTop();
-          }
+      $scope.currentVisiblePanelName = null;
+
+      $scope.isPanelVisible = function(panelName) {
+        if (panelName === $scope.PANEL_TUTOR && $scope.canWindowFitTwoCards()) {
+          return true;
+        } else {
+          return panelName == $scope.currentVisiblePanelName;
+        }
+      };
 
-          var page = $('html, body, iframe');
+      $scope.setVisiblePanel = function(panelName) {
+        $scope.currentVisiblePanelName = panelName;
+        if (panelName === $scope.PANEL_INTERACTION) {
+          $scope.$broadcast('showInteraction');
+        }
+      };
+
+      $scope.resetVisiblePanel = function() {
+        if ($scope.panels.length === 0) {
+          $scope.currentVisiblePanelName = null;
+        } else {
+          $scope.currentVisiblePanelName = $scope.panels[0];
+        }
+      };
 
-          page.on("scroll mousedown wheel DOMMouseScroll mousewheel keyup touchmove", function() {
-            page.stop();
-          });
+      // Changes the currently-active card, and resets the 'show previous
+      // responses' setting.
+      var _navigateToCard = function(index) {
+        $scope.activeCard = $scope.transcript[index];
+        $scope.arePreviousResponsesShown = false;
+        $scope.interactionIsInline = oppiaPlayerService.isInteractionInline(
+          $scope.activeCard.stateName);
 
-          page.animate({
-            'scrollTop': newScrollTop
-          }, 1000, 'easeOutQuad', function() {
-            page.off("scroll mousedown wheel DOMMouseScroll mousewheel keyup touchmove");
-          }).promise().done(postScrollCallback);
-        });
+        _recomputeAndResetPanels();
       };
 
-      var _addNewCard = function(stateName, contentHtml) {
-        $scope.allResponseStates.push({
+      var _addNewCard = function(stateName, contentHtml, interactionHtml) {
+        var interactionIsInline = oppiaPlayerService.isInteractionInline(
+          stateName);
+        var interactionInstructions = (
+          oppiaPlayerService.getInteractionInstructions(stateName));
+
+        var oldStateName = null;
+        if ($scope.transcript.length >= 1) {
+          oldStateName = $scope.transcript[$scope.transcript.length - 1].stateName;
+        }
+
+        oppiaPlayerService.applyCachedParamUpdates();
+        $scope.transcript.push({
           stateName: stateName,
           content: contentHtml,
+          interaction: interactionHtml,
```

Comment:
Might be better name this 'interactionHtml'. The rest of the codebase indicates 'interaction' represents the schema structure of an interaction.